### PR TITLE
S1144: Add repro for #8342

### PR DESCRIPTION
**Pull request by antonioaversa**
Reproduces #8342

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp8.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Tests.Diagnostics
+﻿using System;
+
+namespace Tests.Diagnostics
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp8.cs:3]: I'm surprised you haven't fought your war against block-scoped namespaces for this file 😄

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.cs [hunk 2]
```diff
@@ -420,3 +420,17 @@ class Repro_8348
 {
     [MyAttribute] void PrivateMethodWithAttribute() { } // FN: due to the attribute
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8342
+class Repro_8342
+{
+    [Private1] public void APublicMethod() => APrivateMethodCalledByAPublicMethod();
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.cs:427]: Maybe it's also worth adding a test case with multiple attributes (and list attributes):
```cs
[SomePublicAttribute, Private1]
public void Something() { }

[SomePublicAttribute]
[Private1]
public void Something2() { }
```

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
Looks good, please check out my suggestion before merging!

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp8.cs (hunk 1 above)
Comment (antonioaversa) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.CSharp8.cs:3]: Yes, in this case there are other namespaces, so I could not easily go for it :-)

**Pull request merged**